lib/akasha/command_router.rb

```diff
@@ -1,45 +1,56 @@
-require_relative 'command_router/default_handler'
+require_relative 'command_router/optimistic_transactor'
 require 'corefines/hash'
 
 module Akasha
   # Routes commands to their handlers.
   class CommandRouter
     using Corefines::Hash
 
-    # Raised when no corresponding target can be found for a command.
-    NotFoundError = Class.new(RuntimeError)
-
-    def initialize(routes = {})
-      @routes = routes.flat_map do |command, target|
-        if target.is_a?(Class)
-          { command => DefaultHandler.new(target) }
-        else
-          { command => target }
-        end
-      end
-    end
-
-    # Registers a custom route, specifying either a lambda or a block.
-    # If both lambda and block are specified, lambda takes precedence.
-    def register_route(command, lambda = nil, &block)
-      callable = lambda || block
-      @routes[command] = callable
+    def initialize(**routes)
+      @routes = routes
     end
 
-    # Registers a default route, mapping a command to an aggregate class.
-    # As a result, when `#route!` is called for that command, the aggregate
-    # will be loaded from repository, the command will be sent to the object
-    # to invoke the object's method, and finally the aggregate will be saved.
-    def register_default_route(command, aggregate_class)
-      register_route(command, DefaultHandler.new(aggregate_class))
+    # Registers a handler.
+    #
+    # As a result, when `#route!` is called for that command, the aggregate will be
+    # loaded from repository, the command will be sent to the object to invoke the
+    # object's method, and finally the aggregate will be saved.
+    def register(command, aggregate_class = nil, &block)
+      raise ArgumentError, 'Pass either aggregate class or block' if aggregate_class && block
```

Comment:
maybe something like `expect_exacly_one_argument`? It will raise exceptions like `object null doesn't have call method` when you pass none of them and try to execute code.

Reply:
What is it?

Reply:
I meant to create method for validation, ex. `expect_exacly_one_argument` which checks if there is exacly one callback and raises error otherwise.

Reply:
Would that be more readable? One line => one line + you have to go to the helper method to understand the argument checking logic which is intimately connected to the actual logic in `register`. If you think it's really worth it, I can change it though.

Reply:
Of course less is more readable. XOR would be the best solution, but it looks like this `!!lambda ^ block`, because ruby doesn't not support this operator. 

I can live witout it, leave it :)